lib/spack/spack/filesystem_view.py

```diff
@@ -0,0 +1,494 @@
+            well.
+
+            Should except an `exclude` keyword argument containing a list of
+            regexps that filter out matching spec names.
+
+            This method should make use of `activate_{extension,standalone}`.
+        """
+        raise NotImplementedError
+
+    def activate_extension(self, spec):
+        """
+            Activate (link) an extension in this view.
+        """
+        raise NotImplementedError
+
+    def activate_standalone(self, spec):
+        """
+            Activate (link) a standalone package into this view.
+        """
+        raise NotImplementedError
+
+    def check_active(self, spec):
+        """
+            Check if the given concrete spec is active in this view.
+        """
+        raise NotImplementedError
+
+    def deactivate(self, *specs, **kwargs):
+        """
+            Deactivate given specs in view.
+
+            The supplied spec might be a standalone package or an extension of
+            another package.
+
+            Should accept `with_dependencies` as keyword argument (default
+            True) to indicate wether or not dependencies should be deactivated
+            as well.
+
+            Should except an `exclude` keyword argument containing a list of
+            regexps that filter out matching spec names.
+
+            This method should make use of `deactivate_{extension,standalone}`.
+        """
+        raise NotImplementedError
+
+    def deactivate_extension(self, spec):
+        """
+            Deactivate (unlink) an extension in this view.
+        """
+        raise NotImplementedError
+
+    def deactivate_standalone(self, spec):
+        """
+            deactivate (unlink) a standalone package into this view.
+        """
+        raise NotImplementedError
+
+    def get_all_specs(self):
+        """
+            Get all specs currently active in this view.
+        """
+        raise NotImplementedError
+
+    def get_spec(self, spec):
+        """
+            Return the actual spec linked in this view (i.e. do not look it up
+            in the database by name).
+
+            `spec` can be a name or a spec from which the name is extracted.
+
+            As there can only be a single version active for any spec the name
+            is enough to identify the spec in the view.
+
+            If no spec is present, returns None.
+        """
+        raise NotImplementedError
+
+    def print_status(self, *specs, **kwargs):
+        """
+            Print a short summary about the given specs, detailing whether..
+                * ..they are active in the view.
+                * ..they are active but the activated version differs.
+                * ..they are not activte in the view.
+
+            Takes `with_dependencies` keyword argument so that the status of
+            dependencies is printed as well.
+        """
+        raise NotImplementedError
+
+
+def with_extensions(f):
```

Comment:
Hm… I wanted to mirror the "global" behavior of the `DirectoryLayout`/databases etc. Why should extensions be passed explicitly but everything else remains in global objects residing under `spack.store` - seems a bit asymmetric.

My actual primary motivation was to keep `Package.activated` a property, but you are right, explicit is better than implicit.